studio-docs/source/contracts.mdx

```diff
@@ -91,27 +91,30 @@ type Query {
   topProducts: [Product!]!
 }
 
-type Product @key(fields: "upc") {
+# All fields on the "Product" object inherit the "partner" tag so we can avoid tagging them explicitly
+type Product @key(fields: "upc") @tag(name: "partner") {
   upc: ID!
   name: String!
   description: String!
-  # Tagged fields
+  # The internalId, and percentageMatch fields would also inherit the "partner" tag, so they need
+  # the "private" and "experimental" tag (and an exclude filter) to keep being inaccessible
   internalId: ID! @tag(name: "private")
   percentageMatch: Float! @tag(name: "experimental")
 }
 ```
 
-This schema applies the `@tag` directive to two object fields: `internalId` and `percentageMatch`.
+This schema applies the `@tag` directive to one object: `Product`, and two object fields: `internalId` and `percentageMatch`. 
 
 Each tag has a string `name`. You tag types and fields with the _same_ `name` if they should be included or excluded as a group by a particular contract.
 
-Whenever your source variant's [API schema](https://www.apollographql.com/docs/federation/#federated-schemas) is generated, that schema retains all of the `@tag`s from your subgraph schemas.
+Whenever your source variant's [Supergraph schema](https://www.apollographql.com/docs/federation/#federated-schemas) is generated, that schema retains all of the `@tag`s from your subgraph schemas.
 
 #### Tagging rules
 
 * You can apply tags to the following in your schema:
     * Fields of object types (as shown above)
     * Definitions of object, interface, and union types
+    * Tags on an object or interface are inherited by all of its fields (as shown above)
```

Comment:
What happens when a type is _extended_ by another subgraph and fields are added, and the other subgraph doesn't tag the type?

Reply:
So tags get merged across subgraphs and extended types. So if `A` has tag `public` in subgraph A and extension of `A` has tag `private` in subgraph B, in the supergraph schema all fields on `A` will contain both the `public` and `partner` tag.

Definitely could be more clear on the behavior:
```
Tags applied to objects and interfaces (as well as tags applied to extension types) are inherited by all fields of that type.
```